Update component style doc regarding relative URLs

dariobraun opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
The current documentation does not match the mechanics of angular apps built with the CLI.

Issue Number: #22956


## What is the new behavior?
The documentation gives information in accordance with the up to date way of building angular apps - the CLI. This helps minimize confusion.
Explains how files are referenced when included in CLI built process.
Alternative: Only remove the following line:
`The link tag's href URL must be relative to the application root, not relative to the component file.`

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

dariobraun commented:
Is this still relevant? If so I'd appreciate a new review

mary-poppins commented:
You can preview 0f24b25 at https://pr24471-0f24b25.ngbuilds.io/.

brandonroberts commented on aio/content/guide/component-styles.md:
```diff
@@ -216,11 +216,10 @@ You can also write `<link>` tags into the component's HTML template.
 When building with the CLI, be sure to include the linked style file among the assets to be copied to the server as described in the [CLI documentation](https://github.com/angular/angular-cli/wiki/stories-asset-configuration).
 
+Once included, angular will find the stylesheet, whether the link tag's href URL is relative to the application root or the component file.
+
```
I would update this to `Once included, the CLI will include the stylesheet...`

brandonroberts commented:
@dariobraun I left you some feedback. Will you rebase on master, squash down to a single commit, and change the git commit message to `docs: update component styles doc regarding relative URL`.

Thanks

Commit:
docs: update component styles doc regarding relative URL

Update the documentation to match the CLI mechanics regarding relative URL in link tags.
docs: update info on stylesheet location for CLI

dariobraun commented:
@brandonroberts Thanks for the feedback! I incorporated it :)

mary-poppins commented:
You can preview 5f2bee5 at https://pr24471-5f2bee5.ngbuilds.io/.
